### PR TITLE
Require Precinct Selection on VxScan

### LINKED ISSUE #1972: Nitpick here - inconsistent select a precinct language:
Nitpick here - inconsistent select a precinct language:

Issue automatically created from slack feedback reported by ginny 
https://votingworks.slack.com/archives/CJU9MSC6S/p1655396178153819?thread_ts=1655396178.153819&cid=CJU9MSC6S

### PR DESCRIPTION
**Pull request by adghayes**
## Overview
Closes #1972. Requires that a precinct (or specifically "All Precincts") is selected in the Election Manager menu before the machine is operable or the poll worker screen is accessible.

### VxScan Configure Flow
[must-select-precinct.webm](https://user-images.githubusercontent.com/37960853/187556618-5f27f1a9-c94b-43c7-b017-203bf107cab2.webm)

### VxMark Tweak
Alongside this change, I tweaked the copy in the BMD unconfigure screen so it's apparent when it's unconfigured because there's no election or when there's no precinct. This is as much a tool for me in development and testing as anything.

#### No Election
Screen Shot 2022-08-30 at 3 49 03 PM

#### No Precinct
Screen Shot 2022-08-30 at 3 48 45 PM

## Testing Plan
- Manual testing
- Added test in `app.test.tsx` and cleaned up a related test

## Checklist
- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.
- [ ] I have added JSDoc comments to any newly introduced exports

- [x] I have added a screenshot and/or video to this PR to demo the change
- [x] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

**Comment by adghayes** [comment]
@arsalansufi I took a different approach than your initial pass. I made the assumption that the poll worker screen should never be accessed (and cannot be accessed) if no precinct is selected. This simplified some of the safety checking.

File: frontends/precinct-scanner/src/screens/unconfigured_precinct_screen.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      <CenteredLargeProse>
+        <h1>No Precinct Selected</h1>
+        <p>Insert an Election Manager card to select a precinct.</p>
+      </CenteredLargeProse>
+    </ScreenMainCenterChild>
+  );
+}
+
+/* istanbul ignore next */
+export function DefaultPreview(): JSX.Element {
```
Comment (eventualbuddha) [inline frontends/precinct-scanner/src/screens/unconfigured_precinct_screen.tsx:21]: You need to add to `PreviewApp` too:

```diff
diff --git a/frontends/precinct-scanner/src/preview_app.tsx b/frontends/precinct-scanner/src/preview_app.tsx
index 6af19b8d2..e41a14627 100644
--- a/frontends/precinct-scanner/src/preview_app.tsx
+++ b/frontends/precinct-scanner/src/preview_app.tsx
@@ -25,6 +25,7 @@ import * as ScanJamScreen from './screens/scan_jam_screen';
 import * as ScanBusyScreen from './screens/scan_busy_screen';
 import * as SetupScannerScreen from './screens/setup_scanner_screen';
 import * as UnconfiguredElectionScreen from './screens/unconfigured_election_screen';
+import * as UnconfiguredPrecinctScreen from './screens/unconfigured_precinct_screen';
 import * as ReplaceBallotBagScreen from './components/replace_ballot_bag_screen';
 
 export function PreviewApp(): JSX.Element {
@@ -52,6 +53,7 @@ export function PreviewApp(): JSX.Element {
         ScanBusyScreen,
         SetupScannerScreen,
         UnconfiguredElectionScreen,
+        UnconfiguredPrecinctScreen,
         ReplaceBallotBagScreen,
       ]}
     />
```

**Review by eventualbuddha — COMMENTED** [review]
image

Looks like the bottom bar still says "All Precincts" when nothing is selected.

File: frontends/precinct-scanner/src/screens/poll_worker_screen.tsx [hunk 2]
```diff
@@ -118,16 +121,23 @@ export function PollWorkerScreen({
   const hasPrinterAttached = printerFromProps || !window.kiosk;
   const { election } = electionDefinition;
 
-  const precinct = election.precincts.find((p) => p.id === currentPrecinctId);
+  const precinct =
+    currentPrecinctId === ALL_PRECINCTS_OPTION_VALUE
+      ? ALL_PRECINCTS_OPTION_VALUE
+      : election.precincts.find((p) => p.id === currentPrecinctId);
```
Comment (jonahkagan) [inline frontends/precinct-scanner/src/screens/poll_worker_screen.tsx:127]: You can use the `find` function from @votingworks/utils to automatically assert that the precinct is found

**Comment by adghayes** [comment]
Fix for showing "All Precincts" when there is actually no precicnt:

Screen Shot 2022-08-30 at 5 36 28 PM


Screen Shot 2022-08-30 at 5 36 36 PM

File: frontends/precinct-scanner/src/screens/poll_worker_screen.tsx (hunk 2 above)
Comment (adghayes) [reply frontends/precinct-scanner/src/screens/poll_worker_screen.tsx]: Cool, thanks!

**Review by arsalansufi — APPROVED** [review]
> @arsalansufi I took a different approach than your initial pass. I made the assumption that the poll worker screen should never be accessed (and cannot be accessed) if no precinct is selected. This simplified some of the safety checking.

Great call! I really like how this has simplified things - LGTM!

**Review by eventualbuddha — APPROVED** [review]

**Pull request merged**